pkg/services/services.go

```diff
@@ -77,6 +78,15 @@ func toAddressService(service *v1.ServiceInstance) (result []kclient.Object) {
 		newService.Spec.ExternalName = service.Spec.Address
 	} else {
 		newService.Spec.Type = corev1.ServiceTypeClusterIP
+
+		// The baaah route we are on does not prune Endpoints,
+		// so we need to add this annotation to override it.
+		endpointsAnnotations := make(map[string]string, len(newService.Annotations)+1)
+		for k, v := range newService.Annotations {
+			endpointsAnnotations[k] = v
+		}
+		endpointsAnnotations[apply.AnnotationPrune] = "true"
```

Comment:
Do you want to put these annotations on the `Endpoints` object?

Reply:
Yep that was the goal lol. Updated. Thanks for catching that.